nion/instrumentation/stem_controller.py

```diff
@@ -504,6 +505,17 @@ def get_reference_setting_index(self, settings_control: str) -> typing.Optional[
         """
         raise NotImplementedError()
 
+    def convert_axis(self, value: Geometry.FloatPoint, from_axis: str, to_axis: str) -> Geometry.FloatPoint:
```

Comment:
This method name is not descriptive of what it is doing. It is converting a point from one axis to another; not converting an axis.

I'm also wondering whether we want to be using `axis_id` or the `AxisDescriptor` itself.

Another option is introducing a `Matrix` class to `Geometry.py` and returning the matrix from a function named `get_axis_transform_matrix` or something similar. That would be a more general solution that could handle subtleties of the origin intrinsic to rotations. A `Matrix` class could be [reused elsewhere](https://github.com/nion-software/nionswift/blob/e24c7d20e8bac28eb170cd78c21595f1b2962c62/nion/swift/model/Graphics.py#L99-L107) too.

Finally, what about the idea of making the simply adding a method to `STEMController` called `get_axis_by_id` and returning the `AxisDescriptor` (rename to `Axis`?).

Another option here is having methods of the `AxisDescriptor` to return a `Matrix` to convert to another axis. The downside of this is that it would would involve having a reference axis common to all axes used within the system - but this exists at some level already (in AS2?).

Reply:
This is all way to complicated for what I'm trying to achieve here...I simply want a function that I can call that asks AS2 to convert a vector between two coordinate systems. We don't have any information available in Swift about how different coordinate systems in AS2 are connected and I think moving this from AS2 to Swift would be a huge project and not really worth the effort. As I wrote in the pull-request in `nion-instrumentation`: I'm also fine with only having `convert_axis` available in our Nion-specific implementation of `stem_controller`.